tweet_recommendations/data_processing/data_loader.py

```diff
@@ -1,45 +1,60 @@
+import pickle
+from typing import Dict, List, Tuple
+
 import pandas as pd
 import preprocessor as p
 
 from tweet_recommendations.utils.constants import EXTERNAL_DATA_PATH, OUTPUT_DATA_PATH
 
 
 def load_dataset_as_dataframe_with_given_fields(
-    fields=["hashtags", "text", "retweet_count", "id"], filename="original_tweets.p"
-):
-    tweets_df = pd.read_pickle(EXTERNAL_DATA_PATH.format(filename=filename))
+    fields=("hashtags", "text", "retweet_count", "id")
+) -> pd.DataFrame:
+    tweets_df = pd.read_pickle(EXTERNAL_DATA_PATH)
     dataset = pd.DataFrame(tweets_df)[fields]
     return dataset
 
 
-def save_dataset_as_pickle(dataset, filename="output_dataset.pkl"):
-    dataset.to_pickle(OUTPUT_DATA_PATH.format(filename=filename))
+def save_dataset_as_pickle(dataset: pd.DataFrame):
+    dataset.to_pickle(OUTPUT_DATA_PATH)
+
 
+def convert_hashtags_dicts_to_list(tweets_df: pd.DataFrame) -> pd.DataFrame:
+    assert "hashtags" in tweets_df
+    tweets_df["hashtags"] = tweets_df["hashtags"].apply(
+        lambda x: [item["text"] for item in x]
+    )
+    return tweets_df
 
-def extract_hashtags(list_of_dicts):
+
+def extract_hashtags(list_of_dicts: List[Dict]) -> List[str]:
     output_hashtags = []
     for hash_dict in list_of_dicts:
         output_hashtags.append(hash_dict["text"])
     return list(set(output_hashtags))
 
 
-def tokenize_tweet_content_to_types(dataset, tokenize_type_list):
+def tokenize_tweet_content_to_types(
+    dataset: pd.DataFrame, tokenize_type_list: List[str]
+) -> pd.DataFrame:
     """Tokenize all tweets with with defined contents i.e 'something #DataScience' with 'something $HASHTAG$' """
     tuple_to_unpack = get_filter_obejcts_as_tuple(tokenize_type_list)
     p.set_options(*tuple_to_unpack)
     dataset["text"] = dataset["text"].apply(lambda txt: p.tokenize(txt))
     return dataset
 
 
-def get_cleared_dataset_without_specific_content(dataset, clearing_type_list):
+def get_cleared_dataset_without_specific_content(
+    dataset: pd.DataFrame, clearing_type_list: List[str]
+) -> pd.DataFrame:
     dataset["hashtags"] = dataset["hashtags"].apply(lambda txt: extract_hashtags(txt))
     tuple_to_unpack = get_filter_obejcts_as_tuple(clearing_type_list)
     p.set_options(*tuple_to_unpack)
     dataset["text"] = dataset["text"].apply(lambda txt: p.clean(txt))
     return dataset
 
 
-def get_filter_obejcts_as_tuple(text_content_types):
+def get_filter_obejcts_as_tuple(text_content_types: List[str]):
```

Comment:
Literówka: obejcts -> objects
```suggestion
def get_filter_objects_as_tuple(text_content_types: List[str]):
```